sympy/sets/fancysets.py

```diff
@@ -752,11 +752,12 @@ def _boundary(self):
     def as_relational(self, x):
         """Rewrite a Range in terms of equalities and logic operators. """
         from sympy.functions.elementary.integers import floor
-        i = (x - (self.inf if self.inf.is_finite else self.sup))/self.step
+        from sympy.functions.elementary.piecewise import Piecewise
+        i = (x - Piecewise((self.inf, self.inf.is_finite != False), (self.sup, True)))/self.step
         return And(
             Eq(i, floor(i)),
-            x >= self.inf if self.inf in self else x > self.inf,
-            x <= self.sup if self.sup in self else x < self.sup)
```

Comment:
Seems like you would want `x < oo` instead of `x <= oo`, wouldn't you?

Reply:
Oh I see. Well, other than `self.inf = oo`, this `x <= self.inf` is fine, right?

Reply:
`x>=self.inf, (x<self.sup if self.sup.is_infinite else x<=self.sup)`